### DISCUSSION TITLE
Display asset (name) in a submission overview

### DISCUSSION THREAD

**Discussion by FrDH** [solspace/craft-freeform#1688] [Questions & Answers] [answered]
When trying this guide, is it possible to display anything that was uploaded to an asset/upload field?
https://docs.solspace.com/craft/freeform/v4/guides/dynamic-data/displaying-submission-data-from-previous-pages/

It's worth noting that all assets are uploaded to a "protected" directory, one that does not have a public URL. This can not change.

I tried something like this:
```
{% set value = form.get('myUploadField').value %} 
{# something like 1234456 #}

{% set asset = craft.assets.id(value).one() %} 
{# null #}
```

**Comment by kjmartens** [comment]
Hi @FrDH,

Can you try something like this instead?

```twig
{% set assetId = form.get('myUploadField').value %}
<p>Here's the uploaded file(s):</p>
<ul>
    {% for asset in craft.assets.id(assetId).all %}
        <li>{{ asset.url }}</li>
    {% endfor %}
</ul>
```

Also, are you entering this into a Twig-enabled HTML block in the form builder, or something else? 🙂

**Comment by FrDH** [comment]
Thank you for the suggestion, but unfortunately it does not out output anything (other than an empty UL).
It is indeed in a Twig-enabled HTML block.

I changed the upload-directory to one with a public URL, but that didn't help.

```twig
{% for field in allFields|filter(f => f.valueAsString and f.type in ['file', 'file_drag_and_drop']) %}
 <dt>{{ field.label }}</dt>
 <dd>
     <!-- {{ field.valueAsString }} -->
     {% for asset in craft.assets.id(field.valueAsString).all %}
        url: {{ asset.url }}<br>
    {% endfor %}
     </dd>
{% endfor %}
</dl>
```

Note that `.valueAsString` output a 6 digits number, while trying to print `.value` results in an internal server error.

Thanks for any insights.

**Reply by gustavs-gutmanis** [reply, marked as answer]
Hi @FrDH,

You can access the [`AssetQuery`](https://github.com/solspace/craft-freeform/blob/v4/packages/plugin/src/Fields/FileUploadField.php#L83) directly on the `FileUploadField` via `::getAssets()`.

```twig
<dl>
  {% for field in allFields|filter(f => f.valueAsString and f.type in ['file', 'file_drag_and_drop']) %}
    <dt>{{ field.label }}</dt>
    <dd>
       {% for asset in field.assets.all %}
          url: {{ asset.url }}<br>
       {% endfor %}
    </dd>
  {% endfor %}
</dl>
```

**Comment by FrDH** [comment]
@gustavs-gutmanis Thanks! That works.
Just for future references; is this documented anywhere or should I otherwise have been able to figure this out myself? (dumping/printing the entire `field` variable in the twig-enabled HTML block resulted in an internal server error)

**Reply by gustavs-gutmanis** [reply]
We have missed this in our docs, you shouldn't have to figure this out yourself 🙂 
We will update the docs.
